src/features/content-answers/ContentAnswersModal.tsx

```diff
@@ -76,8 +75,9 @@ const ContentAnswersModal = ({ api, currentUser, file, isOpen, onRequestClose }:
                 const response = await api.getIntelligenceAPI(true).ask(prompt, items);
                 handleSuccessCallback(response, updatedQuestions);
             } catch (e) {
-                handleErrorCallback(e);
+                handleErrorCallback(e, updatedQuestions);
```

Comment:
sorry it might be out of scope but I didn't realize this before: are these "callbacks" necessary?

why not something like:
```
const handleOnAsk = useCallback(
    ...
    let nextQuestion = { prompt };
    setQuestions([...questions, nextQuestion]);
    setIsLoading(true);
    try {
        const response = await api.getIntelligenceAPI(true).ask(prompt, items);
        nextQuestion = {
            answer: response.data.answer,
            createdAt: response.data.created_at,
            prompt,
        };
    } catch (error) {
        nextQuestion = {
            error,
            prompt,
        };
    }
    setQuestions([...questions, nextQuestion]);
    setIsLoading(false);
```
that way you don't need `handleSuccessCallback` and `handleErrorCallback` and avoids the object mutation

Reply:
This is a very great point! I think previously it is written as `handleSuccessCallback` and `handleErrorCallback` to better address which part need to work on for following small tasks. 

When bring everything back to try catch, it still needs to update the last question either in place or pop up and then append it back since we need to append the question with prompt into the array at the very beginning.

Reply:
removed the object mutation logic